perun-cli/setRole

```diff
@@ -2,170 +2,125 @@
 			} else {
-				$authzResolverAgent->setRole( authorizedGroup => $authorizedGroupId, complementaryObject => $vo, role => $role );
+				$authzResolverAgent->setRole(authorizedGroup => $authorizedGroupId,  role => $role);
 			}
-			printMessage("$role role is successfully set for authorizedGroup $authorizedGroupId and VO $voId", $batch);
-		}
-	#resource
-	} elsif (!defined($facilityId) && !defined($groupId) && !defined($voId) && defined($resourceId) && !defined($serviceId)) {
-		my $resourcesAgent = $agent->getResourcesAgent;
-		my $resource = $resourcesAgent->getResourceById( id => $resourceId );
-		if (defined($userId)) {
-			$authzResolverAgent->setRole( user => $userId, complementaryObject => $resource, role => $role );
-			printMessage("$role role is successfully set for user $userId and resource $resourceId", $batch);
-		} else {
-			$authzResolverAgent->setRole( authorizedGroup => $authorizedGroupId, complementaryObject => $resource, role => $role );
-			printMessage("$role role is successfully set for authorizedGroup $authorizedGroupId and resource $resourceId", $batch);
+			my $displayName = $rule->getDisplayName();
+			printMessage("Role $displayName successfully set for group $authorizedGroupId", $batch);
+			exit 0;
 		}
-	#serviceId
-	} elsif (!defined($facilityId) && !defined($groupId) && !defined($voId) && !defined($resourceId) && defined($serviceId)) {
-		my $servicesAgent = $agent->getServicesAgent;
-		my $service = $servicesAgent->getServiceById( id => $serviceId );
-		if (defined($userId)) {
-			$authzResolverAgent->setRole( user => $userId, complementaryObject => $service, role => $role );
-			printMessage("$role role is successfully set for user $userId and service $serviceId", $batch);
-		} else {
-			$authzResolverAgent->setRole( authorizedGroup => $authorizedGroupId, complementaryObject => $service, role => $role );
-			printMessage("$role role is successfully set for authorizedGroup $authorizedGroupId and service $serviceId", $batch);
-		}
-	#no object or more than one
-	} else {
-		die "ERROR: only one of objects (facility,vo,group,resource,service) id need to be set\n";
-}
+	}
 }
 
+die "ERROR: role $role not found\n";
 
+
+sub getObject {
+	switch($compObjectName) {
+		case "Vo" {
+			my $vosAgent = $agent->getVosAgent;
+			return $vosAgent->getVoById( id => $objectId );
+		}
+		case "Facility" {
+			my $facilitiesAgent = $agent->getFacilitiesAgent;
+			return $facilitiesAgent->getFacilityById( id => $objectId );
+		}
+		case "Group" {
+			my $groupsAgent = $agent->getGroupsAgent;
+			return $groupsAgent->getGroupById( id => $objectId );
+		}
+		case "Resource" {
+			my $resourcesAgent = $agent->getResourcesAgent;
+			return $resourcesAgent->getResourceById( id => $objectId );
+		}
+		case "Service" {
+			my $servicesAgent = $agent->getServicesAgent;
+			return $servicesAgent->getServiceById( id => $objectId );
+		}
+		else {
```

Comment:
i think we will soon be working more also with securityAdmin role - worth adding it now to both scripts